mlbench_core/optim/pytorch/optim.py

```diff
@@ -465,3 +465,58 @@ def step(self, closure=None):
         self.agg(self.model, self.agg_mode)
         loss = super(CentralizedAdam, self).step(closure=closure)
         return loss
+
+
+def get_optimizer(optimizer, params=None, lr=required, weight_decay=0, sparse_grad_size=10, random_sparse=False,
```

Comment:
As we add optimizers, we will potentially have more and more arguments, making this argument list very long. I would suggest passing one argument in a dict format, that would be the optimizer

Reply:
What is your idea about the dict format? One argument for the name of the optimizer plus the additional optimizer-specific arguments? Maybe it is best if we have the name of the optimizer as separate argument and then an additional **kwargs argument to allow a variable number of arguments to be passed? I would still keep the possible parameters in the docstring though so that users know what can be passed as an argument. What do you think?

Reply:
> What is your idea about the dict format? One argument for the name of the optimizer plus the additional optimizer-specific arguments? 

Yes that is what i mean, basically just pass the name separately and all the specific params as **kwargs.

I don't think there is need to put everything in the comments, because anyone wanting to change optimizer should go see its documentation.

One way of maybe going around this is just to link each optimizer's doc inside this doc.

Reply:
I implemented the suggested changes in the newest commit.